filesystems/localfilesystem/src/main/kotlin/com/lightteam/localfilesystem/repository/LocalFilesystem.kt

```diff
@@ -210,8 +233,20 @@ class LocalFilesystem(private val defaultLocation: File) : Filesystem {
         return Single.create { emitter ->
             val file = File(fileModel.path)
             if (file.exists()) {
-                val text = file.readText(fileParams.charset)
-                emitter.onSuccess(text)
+                val charset = if (fileParams.chardet) {
+                    CJKCharsetDetector.DEBUG = BuildConfig.DEBUG
+                    file.inputStream().use {
+                        CJKCharsetDetector.detect(it) ?: fileParams.charset
```

Comment:
It works pretty bad with files (with UTF-8) that has Russian letters, can this be improved? It's really important for me because I work with these almost every day

Reply:
Yes， I had improved in new version.